GVRf/Framework/framework/src/main/jni/objects/mesh.cpp

```diff
@@ -207,7 +217,86 @@ void Mesh::bindVertexAttributes(GLuint programID)
     }
     unSetVaoDirty();
 }
+void Mesh::updateVBOs(unsigned int vtx_count, unsigned int idx_start, unsigned int indices_count, std::vector<float>& matrix_indices,
+		const std::vector<glm::vec3>& vertices, std::vector<unsigned short>& indices, const std::vector<glm::vec2>& texcords,
+		const std::vector<glm::vec3>& normals){
+
+	if(vert_vboID_ == GVR_INVALID || norm_vboID_ == GVR_INVALID
+            || tex_vboID_ == GVR_INVALID || triangle_vboID_ ==GVR_INVALID){
+        std::string error = "VAO is not created ";
+        throw error;
+        return;
+    }
```

Comment:
This should not be necessary -- with the attribute interleaving code and https://github.com/Samsung/GearVRf/pull/685 VAO/VBO details/state are internal to the Mesh instance